File: trunk/hangman_js/Guess.js
"use strict"

/**
 * Common interface for GuessWord and GuessLetter
 */
function Guess() {
}

/**
 * Applies the current guess to the specified game.
 * @param game The game to make the guess on.
 */
Guess.prototype.makeGuess = function(game) {
  throw "Not implemented!";
}

exports.Guess = Guess;

File: hangman_lua/Guess.lua
module(..., package.seeall)

cl = require "cl"

Guess = cl.makeClass {
  -- Applies the current guess to the specified game.
  -- @param game The game to make the guess on.
  makeGuess = function(self, game)
    assert(false, "Not implemented!")
  end
}

return Guess
